### PR TITLE
Lib search

### PR DESCRIPTION
**Pull request by sopos**
an ability to search in libs/<component>/<library> path

### CODE REVIEW

**Comment by sopos** [comment]
@psss @lukaszachy, please take a look the the change whether you think this is it

**Comment by kkaarreell** [comment]
Could you please rather consider to extend BEAKERLIB_LIBRARY_PATH usage to specify colon separated list of directories? For fmf it would be trivial to update the variable prior executing the test.

**Comment by sopos** [comment]
now the doc update is needed

File: src/libraries.sh [hunk 1]
```diff
@@ -213,10 +239,25 @@ The library search mechanism is based on Beaker test hierarchy system, i.e.:
 
 /component/type/test-name/test-file
 
-When test-file calls rlImport with 'foo/bar' parameter, the directory path
-is traversed upwards, and a check for presence of the test /foo/Library/bar/
-will be performed. This means this function needs to be called from
-the test hierarchy, not e.g. the /tmp directory.
+When test-file calls rlImport with 'foo/bar' parameter, the lilraries are search
```
Comment (psss) [inline src/libraries.sh:242]:
```suggestion
When test-file calls rlImport with 'foo/bar' parameter, the libraries are searched
```

File: src/libraries.sh [hunk 2]
```diff
@@ -213,10 +239,25 @@ The library search mechanism is based on Beaker test hierarchy system, i.e.:
...
+
+the next component of the path may be:
+
+    - /foo/Library/bar
+    - /*/foo/Library/bar
+    - /libs/foo/bar
+
+the directory path is then constructed as prefix/path/lib.sh
+If the library is still not found a upwards directory traversal is used, and a
+check for presence of the test /foo/Library/bar/ or libs/foo/bar/ is be
```
Comment (psss) [inline src/libraries.sh:258]:
```suggestion
check for presence of the test /foo/Library/bar/ or libs/foo/bar/ is to be
```

File: src/libraries.sh [hunk 3]
```diff
@@ -213,10 +239,25 @@ The library search mechanism is based on Beaker test hierarchy system, i.e.:
...
+    - /usr/share/beakerlib-libraries
+
+the next component of the path may be:
+
+    - /foo/Library/bar
+    - /*/foo/Library/bar
+    - /libs/foo/bar
+
+the directory path is then constructed as prefix/path/lib.sh
+If the library is still not found a upwards directory traversal is used, and a
```
Comment (psss) [inline src/libraries.sh:257]:
```suggestion
If the library is still not found an upwards directory traversal is used, and a
```

File: src/libraries.sh [hunk 4]
```diff
@@ -213,10 +239,25 @@ The library search mechanism is based on Beaker test hierarchy system, i.e.:
...
-the test hierarchy, not e.g. the /tmp directory.
+When test-file calls rlImport with 'foo/bar' parameter, the libraries are searched
+in following locations:
+these are the possible path prefixes
+
+    - colon-separated paths from $BEAKERLIB_LIBRARY_PATH
+    - /mnt/tests
+    - /usr/share/beakerlib-libraries
+
+the next component of the path may be:
```
Comment (sopos) [inline src/libraries.sh:250]:
```suggestion
the next component of the path is one of the following:
```

File: src/libraries.sh [hunk 5]
```diff
@@ -213,10 +239,25 @@ The library search mechanism is based on Beaker test hierarchy system, i.e.:
...
+
+the next component of the path is one of the following:
+
+    - /foo/Library/bar
+    - /*/foo/Library/bar
+    - /libs/foo/bar
+
+the directory path is then constructed as prefix/path/lib.sh
+If the library is still not found an upwards directory traversal is used, and a
+check for presence of the test /foo/Library/bar/ or libs/foo/bar/ is be
```
Comment (sopos) [inline src/libraries.sh:258]:
```suggestion
check for presence of the library in /foo/Library/bar/ or libs/foo/bar/ is be
```

File: src/libraries.sh [hunk 6]
```diff
@@ -213,10 +239,25 @@ The library search mechanism is based on Beaker test hierarchy system, i.e.:
...
+
+the next component of the path is one of the following:
+
+    - /foo/Library/bar
+    - /*/foo/Library/bar
+    - /libs/foo/bar
+
+the directory path is then constructed as prefix/path/lib.sh
+If the library is still not found an upwards directory traversal is used, and a
+check for presence of the test /foo/Library/bar/ or libs/foo/bar/ is to be
```
Comment (sopos) [inline src/libraries.sh:258]:
```suggestion
check for presence of the library in /foo/Library/bar/ or libs/foo/bar/ is to be
```

**Pull request merged**